src/shared/lib/accounts/plugins/aws-kms/index.ts

```diff
@@ -0,0 +1,194 @@
+import * as z from "zod";
+
+import { defineAccountPlugin } from "../../account-service";
+import { KMSClient, CreateKeyCommand } from "@aws-sdk/client-kms";
+import { getAwsKmsAccount } from "./get-aws-kms-account";
+import { decrypt } from "../../../../utils/crypto";
+
+const awsKmsCredentialSchema = z.object({
+  accessKeyId: z.string(),
+  secretAccessKey: z.string(),
+});
+
+const awsKmsLegacyConfigSchema = z.object({
+  awsAccessKeyId: z.string().nullable(),
+  awsSecretAccessKey: z.string().nullable(),
+  awsRegion: z.string().nullable(),
+});
+
+type AwsKmsLegacyConfig = z.infer<typeof awsKmsLegacyConfigSchema>;
+type AwsKmsCredential = z.infer<typeof awsKmsCredentialSchema>;
+
+// AWS KMS Plugin Definition
+export const awsKmsPlugin = defineAccountPlugin({
```

Comment:
i was expecting to see a plugin for GCP and one for private key as well? is it just not implemented yet?

Reply:
yep, need to extract the exiting code into this plugin format. Will also be adding one for circle